### ISSUE TITLE
Parentheses not allowed in break statements

### ISSUE THREAD

**Issue by Parent5446** [facebook/hhvm#854]
PHP allows the break and continue statements to take an operand representing the number of levels to continue or break. This operand can be an expression (although it cannot contain variables as of PHP 5.4). In other words, all of the following are valid syntax:

```
break 2;
break(2);
break (1 + 1);
```

However, on the latter two cases, HipHop throws a parse error: HipHop Fatal error: syntax error, unexpected '(', expecting T_LNUMBER or ';'

In other words, it only supports a constant number rather than a numerical expression. This makes it impossible to run certain versions of PHPUnit using HHVM.
